### ISSUE TITLE
RFC: Integrate eth-block-tracker for subscribable queries

### ISSUE THREAD

**Issue by danfinlay** [ethjs/ethjs-contract#2]
## Description

What if ethjs queries returned `promEvents`, somewhat like `Web3@1.0` plans to have.

In addition to calling `.then()` on a result, you could subscribe to updates.  Imagine:

```
token.balanceOf(my_address).on('latest', updateBalanceView)
```

Should be easier than ever with [eth-block-tracker](https://github.com/kumavis/eth-block-tracker).

**Comment by SilentCicero** [comment]
oooooo. We can do this. But I would make in it's own module, and have hooks for it in different modules.

**Comment by danfinlay** [comment]
Yeah maybe it could be like `ethjs-event-contract`, and its initialization takes an `ethjs-contract` and an `eth-block-tracker`, and allows subscribing to updates on any property `constant` function of that contract, as well as maybe access to a synchronous property cache, of last-retrieved values?

**Comment by SilentCicero** [comment]
Yes, that could work and would reduce bulk/module complexity on the main contract object.
